### ISSUE TITLE
[BUG] Toggling cellular data settings causes app to crash

### ISSUE THREAD

**Issue by paladincorners** [owncloud/ios-app#1299]
### Steps to reproduce
1. Open OwnCloud
2. Login in to an account (works fine if you change the settings from the first screen without logging in)
3. Go to settings for that account
4. Toggle cellular data on or off (happens for any of the sub-settings too)
5. App crashes

### Expected behavior
Tell us what should happen
App should not crash when toggling cellular settings

### Actual behavior
Tell us what happens instead
App crashed

### Client
**iOS version:** 

**ownCloud app version:** 12.03

**Device model:**

### Server configuration
**Web server:**

**Database:**

**PHP version:**

**ownCloud version:**

**Storage backend (external storage):**

#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
```

**Comment by jesmrec** [comment]
Reproducible but not every time. Subsettings do not crash on my side. Just reproducible if accounts attached

Some logs caught: 

[ownCloud_17_Nov_2023_at_12_22_28.log.txt](https://github.com/owncloud/ios-app/files/13391022/ownCloud_17_Nov_2023_at_12_22_28.log.txt)

**Comment by michaelstingl** [comment]
> Some logs caught:
> 
> [ownCloud_17_Nov_2023_at_12_22_28.log.txt](https://github.com/owncloud/ios-app/files/13391022/ownCloud_17_Nov_2023_at_12_22_28.log.txt)

Share the crash log from `Settings > Privacy > Analytics > Analytics Data`

**Comment by jesmrec** [comment]
> Share the crash log from Settings > Privacy > Analytics > Analytics Data

i guess you mean this one:

[ownCloud-2023-11-17-122240.ips.zip](https://github.com/owncloud/ios-app/files/13391887/ownCloud-2023-11-17-122240.ips.zip)

**Comment by felix-schwarz** [comment]
Fixed in the SDK, an updated version of which is included in the latest #1274.

**Comment by jesmrec** [comment]
Fixed.

To be released in 12.1.
